aula-pratica-04/ex6.c

```diff
@@ -0,0 +1,36 @@
+/*
+    Faça uma função que leia uma string de 50 posições digitada por um usuário,
+    e retorne valor 1 se for um palíndromo ou 0 em caso negativo.
+    Um palíndromo é uma palavra, frase ou qualquer outra sequência que tenha
+    a propriedade de poder ser lida tanto da direita para a esquerda como da esquerda
+    para a direita sem que se altere seu significado 
+    (4334 e arara são exemplos de palíndromos).
+*/
+
+#include<stdio.h>
+#include<string.h>
+
+int eh_palindrome(char str[50]) {
+    int size = strlen(str) - 1;
+    char str_reversa[size - 1];
+    for(int i = size; i >= 0; i--) {
+        str_reversa[size - i] = str[i];
```

Comment:
(opcional) se gente fizer aquele lance de inverter string numa função separada, dava pra reutilizar ela aqui, neh? kk

anyway, se não for o caso, e quiser fazer essa lógica inteira aqui, acho que da pra economizar um for de novo, se a gente fizar assim:

```suggestion
        if (str[size - i] != str[i]) {
            return false;
        };
```

Que você acha?

Reply:
Não entendi muito bem essa parte kkk :sweat_smile:

Reply:
é que tipo você esta criando uma string inversa a outra string só pra testar se as duas são iguais, certo?

Você ao invés de criar a string invertida, pode checar se a string é igual ao seu inverso correndo o array tanto numa direção quanto na outra ao mesmo tempo

na primeira iteração o que vai ser testado vai ser:
```
[0][1][2][3][4]
[a][r][a][r][a]
 ^           ^
 i           size - i
```
aí quando i tiver o valor de 1:
```
[0][1][2][3][4]
[a][r][a][r][a]
    ^     ^
    i     size - i
```
e assim por diante

Alias, agora parando pra ver melhor, você pode parar de correr o array na metade da string kkk

Reply:
tendeu?

Reply:
Ah sim, agora eu visualizei a lógica, faz mais sentido mesmo